validation/parser/pipeline/parser_pipeline_funcs.py

```diff
@@ -0,0 +1,32 @@
+from parser.pair import Pair
+from typing import Iterable
+from typing import Iterator
+
+from calculator.distance_calculator import DistanceCalculator
+
+
+def fill_empty(pairs: Iterable[Pair], embedding_size: int) -> Iterator[Pair]:
+    empty_embedding = [[0] * embedding_size]
+    return (Pair(pair.image1 or empty_embedding,
+                 pair.image2 or empty_embedding,
+                 pair.is_match) for pair in pairs)
+
+
+def remove_empty(pairs: Iterable[Pair]) -> Iterator[Pair]:
+    return (pair for pair in pairs if pair.image1 and pair.image2)
+
+
+def filter_target(pairs: Iterable[Pair],
+                  distance_metric: str) -> Iterator[Pair]:
```

Comment:
Why not return an `Iterable`? The implementation can stay the same and the return type can be changed to `Iterable` with mypy still passing. Having the distinction between `Iterable` and `Iterator` doesn't add much in terms of type safety (since all we usually care about is "this thing can be looped over") but it introduces an additional concept to understand.